### PR TITLE
Issue 382: Filters for History Page

### PR DESCRIPTION
**Pull request by Niicck**
We can now filter by date range and community on the History Page.

screen shot 2018-11-30 at 9 33 09 am

Related Issue: #382 
Test it at: http://ctxfloods-frontend-382-history.s3-website-us-east-1.amazonaws.com/dashboard/crossings/history

Backend PR: https://github.com/cityofaustin/ctxfloods-backend/pull/50

### CODE REVIEW

File: src/components/Dashboard/CrossingStatusHistory/CrossingStatusHistoryFilter.js [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+      dateUpperBoundKey,
+      inclusiveEndDate,
+    } = this.state;
+
+    let allCommunities;
+    const communitiesLoading = !this.props.AllCommunitiesQuery || this.props.AllCommunitiesQuery.loading;
+    if (communitiesLoading) {
+      return <div>Loading...</div>
+    }
+    allCommunities = this.props.AllCommunitiesQuery.allCommunities.nodes.filter(node => node.id !== 1337 && node.id !== 9002);
```
Comment (briaguya) [inline src/components/Dashboard/CrossingStatusHistory/CrossingStatusHistoryFilter.js]: not the biggest fan of having 1337 and 9002 in here as magic numbers but it def makes sense for the moment. Worth noting that we're using them here at least.

File: src/components/Dashboard/CrossingStatusHistory/CrossingStatusHistoryItem.js [hunk 2]
```diff
@@ -80,20 +79,19 @@ class CrossingStatusHistoryItem extends Component {
...
+      statusId,
+      statusName,
+      statusReasonName,
+      reopenDate,
+      indefiniteClosure,
+      createdAt,
+      crossingId,
+      crossingName,
+      notes
+    } = update;
```
Comment (briaguya) [inline src/components/Dashboard/CrossingStatusHistory/CrossingStatusHistoryItem.js]: hooray!

File: src/index.scss [hunk 3]
```diff
@@ -29,9 +29,11 @@ body {
 
 .FloodsApp {
   margin: 0 auto;
-  height: calc(100% - 1px);
+  height: 100%; //previously calc(100% - 1px)
```
Comment (briaguya) [inline src/index.scss]: definitely worth verifying this doesn't break mobile safari, I'm pretty sure this led to the page trying to scroll all the time and making it 1px smaller made it not try to scroll.

File: src/components/Dashboard/CrossingStatusHistory/CrossingStatusHistoryFilter.js (hunk 1 above)
Comment (Niicck) [reply src/components/Dashboard/CrossingStatusHistory/CrossingStatusHistoryFilter.js]: Yeah there's one other place where I'm doing that too. It's a temporary measure until I figure out a plan for Super_Admin_World and ALL.

File: src/index.scss (hunk 3 above)
Comment (Niicck) [reply src/index.scss]: Checked - mobile Safari works fine

**Pull request merged**